code/ui/components/dragdrop/Drag.cs

```diff
@@ -0,0 +1,239 @@
+using System.Numerics;
+
+using Sandbox;
+using Sandbox.UI;
+
+// TODO use M4x4 transform
+
+namespace TTTReborn.UI
+{
+    public partial class Drag : DragDrop
+    {
+        public bool IsDragging
+        {
+            get => _isDragging;
+            private set
+            {
+                _isDragging = value;
+
+                if (value == false)
+                {
+                    DragBasePanel.Style.Position = _oldPositionMode;
+
+                    _oldPositionMode = null;
+                }
+                else
+                {
+                    if (_oldPositionMode == null)
+                    {
+                        _oldPositionMode = DragBasePanel.Style.Position;
+                    }
+
+                    DragBasePanel.Style.Position = PositionMode.Absolute;
+                }
+            }
+        }
+        private bool _isDragging = false;
+
+        public bool IsFreeDraggable
```

Comment:
What does this `bool` mean? That the component can be dragged around? Not sure if I understand.

Reply:
That the component can be dragged without limitation (it does not need to snap into the parent, it can stay where you dragged it)

Reply:
sounds fine then, thanks.